### PR TITLE
Cleanups and additional tests to improve test coverage

### PR DESCRIPTION
**Pull request by greglandrum**
No actual new functionality here, this is just some code cleanup and additional testing to improve test coverage on the C++ side.
The new coverage with these changes is 94.4% on lines of code. The original coverage (after merging the fixes for #2814 and #2815, which I discovered while doing this exercise and which already helped things) was 90.4%. Not a huge improvement, but certainly a step forward.

### CODE REVIEW

File: Code/GraphMol/SmilesParse/CMakeCache.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# This is the CMakeCache file.
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/CMakeCache.txt]: I don't think you meant to check in the CMakeCache.txt file

File: Code/GraphMol/SmilesParse/SmilesParse.h [hunk 2]
```diff
@@ -127,12 +127,9 @@ inline std::unique_ptr<RDKit::RWMol> operator"" _smiles(const char *text,
 inline std::unique_ptr<RDKit::RWMol> operator"" _smarts(const char *text,
                                                         size_t len) {
   std::string smi(text, len);
-  RWMol *ptr = nullptr;
-  try {
-    ptr = SmartsToMol(smi);
-  } catch (const RDKit::MolSanitizeException &) {
-    ptr = nullptr;
-  }
+  // no need for exception handling here: SmartsToMol() doesn't do
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/SmilesParse.h]: This comment might only make sense in the context of removing the exception handling :)

File: Code/GraphMol/SmilesParse/SmilesParseOps.cpp [hunk 3]
```diff
@@ -88,21 +88,14 @@ void CleanupAfterParseError(RWMol *mol) {
...
-                  Bond::BondDir bondDir, bool closeRings, bool doingQuery) {
+                  Bond::BondDir bondDir) {
   PRECONDITION(mol, "no molecule");
   PRECONDITION(frag, "no fragment");
   PRECONDITION(mol->getActiveAtom(), "no active atom");
   Atom *lastAt = mol->getActiveAtom();
   int nOrigAtoms = mol->getNumAtoms();
   int nOrigBonds = mol->getNumBonds();
 
-  //
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/SmilesParseOps.cpp]: I'm assuming that closeRings was never actually. being used.

File: Code/GraphMol/SmilesParse/catch_tests.cpp.sav [hunk 4]
```diff
@@ -0,0 +1,406 @@
+//
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/catch_tests.cpp.sav]: Looks like something that was mistakenly checked in

File: Code/GraphMol/SmilesParse/SmilesParseOps.cpp (hunk 3 above)
Comment (bp-kelley) [reply Code/GraphMol/SmilesParse/SmilesParseOps.cpp]: Looks like I was right :)

File: Code/GraphMol/SmilesParse/SmilesParse.h (hunk 2 above)
Comment (greglandrum) [reply Code/GraphMol/SmilesParse/SmilesParse.h]: It's there to make it clear that not having exception handling here (unlike in the `_smiles` version) wasn't an oversight.

**Comment by greglandrum** [comment]
@bp-kelley Thanks for catching those incorrectly added files. They're now gone.

**Comment by greglandrum** [comment]
@bp-kelley : ok to merge this?

**Pull request merged**